### ISSUE TITLE
[Plugin GoogleAds_IMA] Not displayed ad

### ISSUE THREAD

**Issue by gbrn** [WWBN/AVideo#9165]
@DanielnetoDotCom I'm running tests with the GoogleAds_IMA Plugin. But I haven't been successful. The ad is not displayed at the beginning of the videos.

The plugin configuration is as follows:
![image](https://github.com/WWBN/AVideo/assets/37880073/ce9d5fc0-5773-4fbd-9117-bfc68e258399)

I'm using the following URL for testing:
https://basil79.github.io/vast-sample-tags/pg/vast.xml

Any additional configuration that needs to be done?

**Comment by mankota** [comment]
I have similar issue. I have not been able to make the ads display at all.

**Comment by DanielnetoDotCom** [comment]
Please check this https://github.com/WWBN/AVideo/wiki/GoogleAds_IMA---Videos-Ads-on-your-page#troubleshooting

**Comment by gbrn** [comment]
@DanielnetoDotCom The problem is not the URL or the server. This URL I'm using is functional for testing, I checked it and it's working: (https://basil79.github.io/vast-sample-tags/pg/vast.xml).
I also tested the URL you sent and it doesn't work: (https://vast.forjesus.tv/vast.xml).

The problem is with the plugin or its configuration.

I checked the LOG:
` [27-Jun-2024 11:17:52 America/Sao_Paulo] AVideoLog::DEBUG: showAds: CustomizeAdvanced said NOT to show ads on 78 SCRIPT_NAME: /view/videoEmbeded.php`

In the CustomizeAdvanced Plugin, there is nothing related to ADS as the log says. But when accessing the video edit, the option to not display advertising in the video appears activated, which I did not activate.
All videos look like this:

![image](https://github.com/WWBN/AVideo/assets/37880073/10ff7713-d898-4822-98f8-2b4f4d10bc56)

I ask... how do you disable this function of not displaying by default? I did not find it.

**Comment by DanielnetoDotCom** [comment]
So do you mean you cannot uncheck this?

the default is always unchecked

**Comment by gbrn** [comment]
I can uncheck it. The default is unchecked.
I've already unchecked it and still the URL you sent doesn't work (https://vast.forjesus.tv/vast.xml)

Any settings I should check?

**Comment by DanielnetoDotCom** [comment]
Make sure it is unckecked, and try again. 
check your avideo logs, use the https://vast.forjesus.tv/vast.xml for testing and if you can share a video page where the ads should appear

**Comment by gbrn** [comment]
- I made sure it is unchecked and tried again
- Nothing important was recorded in the LOG regarding ADs
- I am using the URL provided by you

Is there anything in this configuration that should be changed?
![image](https://github.com/WWBN/AVideo/assets/37880073/0597a9b8-66f8-44b7-b66c-5a64d1af762c)


Check out the video of the sequence below:
https://github.com/WWBN/AVideo/assets/37880073/677595ff-548e-4a39-a1af-2e86d973345b

Using the following tool I tested that your URL is functional: https://adplayer.pro/developer/vast_inspector
The URL I used previously is also working. So the problem is not the AD server used.
![image](https://github.com/WWBN/AVideo/assets/37880073/1bfd109a-adb2-40b6-b291-c458975d47fd)

LOG:
```
 [27-Jun-2024 17:42:07 UTC] AVideoLog::DEBUG: getUserDbFromUser advancedCustomUser is empty  SCRIPT_NAME: /objects/videoAddViewCount.json.php
 [27-Jun-2024 17:42:07 UTC] AVideoLog::DEBUG: user::recreateLoginFromCookie: user cookie found: myemail@mydomain.com result:  SCRIPT_NAME: /objects/videoAddViewCount.json.php
 [27-Jun-2024 17:41:23 UTC] AVideoLog::DEBUG: user::recreateLoginFromCookie: do cookie-login: myemail@mydomain.com id: 119 SCRIPT_NAME: /plugin/GoogleAds_IMA/VMAP.php
 [27-Jun-2024 17:41:23 UTC] AVideoLog::DEBUG: user::recreateLoginFromCookie: resp=0 SCRIPT_NAME: /plugin/GoogleAds_IMA/VMAP.php
 [27-Jun-2024 17:41:23 UTC] AVideoLog::DEBUG: User:login finish with success users_id= 119 Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 IP=45.173.144.187 SCRIPT_NAME: /plugin/GoogleAds_IMA/VMAP.php
 [27-Jun-2024 17:41:23 UTC] AVideoLog::DEBUG: setUserCookieCredentials there is already a cookie and is the same user SCRIPT_NAME: /plugin/GoogleAds_IMA/VMAP.php
 [27-Jun-2024 17:41:23 UTC] AVideoLog::DEBUG: setUserCookieCredentials there is already a cookie SCRIPT_NAME: /plugin/GoogleAds_IMA/VMAP.php
 [27-Jun-2024 17:41:23 UTC] AVideoLog::DEBUG: User::find line= 1647 SCRIPT_NAME: /plugin/GoogleAds_IMA/VMAP.php
 [27-Jun-2024 17:41:23 UTC] AVideoLog::DEBUG: myemail@mydomain.com user found myemail@mydomain.com SCRIPT_NAME: /plugin/GoogleAds_IMA/VMAP.php
 [27-Jun-2024 17:41:23 UTC] AVideoLog::DEBUG: getUserDbFromUser SELECT * FROM users WHERE user = ?  OR email = ?  LIMIT 1 myemail@mydomain.com SCRIPT_NAME: /plugin/GoogleAds_IMA/VMAP.php
 [27-Jun-2024 17:41:23 UTC] AVideoLog::DEBUG: getUserDbFromUser will also check the email myemail@mydomain.com  SCRIPT_NAME: /plugin/GoogleAds_IMA/VMAP.php
 [27-Jun-2024 17:41:23 UTC] AVideoLog::DEBUG: user::recreateLoginFromCookie: user cookie found: myemail@mydomain.com result:  SCRIPT_NAME: /plugin/GoogleAds_IMA/VMAP.php
```

**Comment by DanielnetoDotCom** [comment]
share the video page where it supposed to have an ad

**Comment by gbrn** [comment]
Danny, it's been resolved...
I was forgetting to do a simple test...
I just noticed that there is a problem with my browser.
I opened an incognito tab without extensions and the ad was displayed.

I use 1 adblock and a Privacy Badger. Even though both are disabled for my domain, there is an AD block.
I found it very strange that this was happening!

It is important to put this detail on WIKI, as it is very simple and goes unnoticed.

![image](https://github.com/WWBN/AVideo/assets/37880073/590a070c-0137-408d-a01b-ebfb71c02e21)
![image](https://github.com/WWBN/AVideo/assets/37880073/c527ff19-7232-4627-8816-628ef395763e)

**Comment by DanielnetoDotCom** [comment]
You may want to use our free plugin AdBlockerDetector

**Comment by gbrn** [comment]
Yes, I checked the AdBlockerDetector plugin, I will be deploying this after the GoogleAds_IMA Plugin.

I'm having trouble finding a VAST ad provider. Google's requirements are very high.
Could you recommend me a Brazilian VAST supplier that has ads in Portuguese?

**Comment by DanielnetoDotCom** [comment]
Hi

Sorry, currently I do not have any one to recommend.

**Comment by gbrn** [comment]
@DanielnetoDotCom Danny, I was able to obtain the VAST URL for video advertising, but it's showing this error.
I can't understand this problem... is the AD player different from the video player?
Could you detail this incompatibility so I can discuss it with my supplier?

![image](https://github.com/user-attachments/assets/94cde867-5e69-4a9c-91ea-924550e39a59)

**Comment by DanielnetoDotCom** [comment]
this looks like you are using out [Ad_Server](https://github.com/WWBN/AVideo/wiki/Ad-Server-Plugin), and you must have a mp4 format for video ads

**Comment by gbrn** [comment]
@DanielnetoDotCom Danny, there really was a wrong configuration in Ad_Server.
I'm running new tests. Thanks.

**Comment by gbrn** [comment]
@DanielnetoDotCom Danny, getting back to testing GoogleAds_IMA.
I can't view VAST videos from the following URL:
`https://pubads.g.doubleclick.net/gampad/ads?iu=/211182487,22738971863/www.cnetv.com.br_Video&description_url=[placeholder]&tfcd=0&npa=0&sz=480x320&gdfp_req=1&unviewed_position_start=1&output=vast&env=vp&impl=s&correlator=`

Can you test it for me?

**Comment by DanielnetoDotCom** [comment]
to use doubleclick URLs you need the GoogleAds_IMA plugin

**Comment by gbrn** [comment]
@DanielnetoDotCom Yes, I'm using it... but for me the ads don't load.
Could you test this URL on your system?

![image](https://github.com/user-attachments/assets/027a36d7-c693-481e-b692-b8b8dad94265)
![image](https://github.com/user-attachments/assets/c0c2d34a-9136-43a9-9f36-5908568a6ae3)

**Comment by DanielnetoDotCom** [comment]
I cannot test your tag, if you want to make sure it is working properly you can try with this tag 

https://vast.forjesus.tv/vast.xml

this will always return ads. 

I cannot guarantee google will provide you with ads
